docker/Dockerfile

```diff
@@ -17,51 +30,118 @@ RUN apt-get update && apt-get install -y libgeos-dev \
 # Install the BCMath PHP extension.
 RUN docker-php-ext-install bcmath
 
-# Inherit from the official Drupal 10 image.
-FROM drupal:10.1
+# Setup dependencies and sources for composer installations.
+FROM baseimage as composer-file
 
-# Set the farmOS and composer project repository URLs and versions.
-ARG FARMOS_REPO=https://github.com/farmOS/farmOS.git
-ARG FARMOS_VERSION=3.x
-ARG PROJECT_REPO=https://github.com/farmOS/composer-project.git
-ARG PROJECT_VERSION=3.x
+# Set the COMPOSER environment variables.
+ENV COMPOSER_HOME=tmp \
+    # Set the COMPOSER_MEMORY_LIMIT environment variable to unlimited.
+    COMPOSER_MEMORY_LIMIT=-1 \
+    # Allow root to install plugins.
+    COMPOSER_ALLOW_SUPERUSER=1
+
+# Install apt dependencies.
+RUN apt-get update && apt-get install -y \
+    # Install git and unzip (needed by Composer).
+    git unzip
```

Comment:
`git` and `unzip` are currently installed and available only in stages related to building farmOS. In my opinion that they should also be available at least in the final dev version of the image but probably also in the final prod image since they are required to install additional modules using composer which is a workflow available in current image versions, so removing it here would be a breaking change.

Reply:
Also, when installing them in the stage that will be included in the final image, it would be good to add `--no-install-recommends` option to the `apt-get install` command to skip installing additional packages that are not really needed and only take up space in the image. It is a good practice to use this option in all instances of apt-get in docker images.

Reply:
I was thinking of community modules, not included in the farmOS itself. For example https://github.com/symbioquine/farmOS_asset_link or https://www.drupal.org/project/farm_map_custom_layers. 

On the [Building farmOS with Composer](https://farmos.org/hosting/composer/#docker) documentation page, there is described approach for adding additional community modules to the composer.json and building docker image based on official production one with these customizations. Currently, in instructions for building a custom image when it comes to installing composer dependencies only `composer install` is provided, without git and unzip in the prod image these instructions will no longer work and will need to be updated to also include installation of git and unzip. So not including git and uzip in production image may break some existing setups when the new farmOS base image version will be pulled.

I would include git and unzip in both dev and prod image to don't break existing documented functionality.

Reply:
Ok, all tough, I think that it would be more appropriate to add the information on the documentation, so the image is kept as slim as possible, It's the right thing to keep it working as it was.
Thanks for the explanation.